### PR TITLE
updated java wrapper for new C API

### PR DESCRIPTION
**Pull request by yanqingmen**

### CODE REVIEW

**Comment by tqchen** [comment]
minor comment:
- please change naming from XgboostError->XGBoostError
- Can you add another build grid in .travis.yml to build the java version and run testcases? 
    - See https://travis-ci.org/dmlc/xgboost and .travis.yml in the root path

**Comment by yanqingmen** [comment]
hi, @tqchen , i've done the modification.

File: scripts/travis_java_script.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Test java package of xgboost
```
Comment (tqchen) [inline scripts/travis_java_script.sh]: add ```set -e ```

**Comment by yanqingmen** [comment]
ok

**Pull request merged**